lib/iris/plot.py

```diff
@@ -529,7 +533,11 @@ def _ensure_cartopy_axes_and_determine_kwargs(x_coord, y_coord, kwargs):
         cartopy_proj = ccrs.PlateCarree()
 
     # Ensure the current axes are a cartopy.mpl.GeoAxes instance.
-    _replace_axes_with_cartopy_axes(cartopy_proj)
+    if 'axes' not in kwargs:
+        _replace_axes_with_cartopy_axes(cartopy_proj)
+    elif not not isinstance(kwargs['axes'], cartopy.mpl.geoaxes.GeoAxes):
```

Comment:
Looks like some of the tests are failing because of this too:

```
======================================================================

ERROR: test_outline (iris.tests.test_mapping.TestLimitedAreaCube)

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/Iris-1.10.0.dev0-py2.7-linux-x86_64.egg/iris/tests/test_mapping.py", line 240, in test_outline

    iplt.outline(self.cube)

  File "/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/Iris-1.10.0.dev0-py2.7-linux-x86_64.egg/iris/plot.py", line 845, in outline

    antialiased=True, coords=coords, axes=axes)

  File "/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/Iris-1.10.0.dev0-py2.7-linux-x86_64.egg/iris/plot.py", line 244, in _draw_2d_from_bounds

    cube, plot_defn, *args, **kwargs)

  File "/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/Iris-1.10.0.dev0-py2.7-linux-x86_64.egg/iris/plot.py", line 614, in _map_common

    return plotfn(*new_args, **kwargs)

  File "/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/matplotlib/pyplot.py", line 2946, in pcolormesh

    ret = ax.pcolormesh(*args, **kwargs)

  File "/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/matplotlib/axes.py", line 7766, in pcolormesh

    t = t._as_mpl_transform(self.axes)

  File "lib/cartopy/_crs.pyx", line 213, in cartopy._crs.CRS._as_mpl_transform (lib/cartopy/_crs.c:4071)

ValueError: Axes should be an instance of GeoAxes, got <class 'matplotlib.axes.AxesSubplot'>

======================================================================

ERROR: test_grid (iris.tests.test_mapping.TestBoundedCube)

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/Iris-1.10.0.dev0-py2.7-linux-x86_64.egg/iris/tests/test_mapping.py", line 205, in test_grid

    iplt.outline(self.cube)

  File "/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/Iris-1.10.0.dev0-py2.7-linux-x86_64.egg/iris/plot.py", line 845, in outline

    antialiased=True, coords=coords, axes=axes)

  File "/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/Iris-1.10.0.dev0-py2.7-linux-x86_64.egg/iris/plot.py", line 244, in _draw_2d_from_bounds

    cube, plot_defn, *args, **kwargs)

  File "/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/Iris-1.10.0.dev0-py2.7-linux-x86_64.egg/iris/plot.py", line 614, in _map_common

    return plotfn(*new_args, **kwargs)

  File "/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/matplotlib/pyplot.py", line 2946, in pcolormesh

    ret = ax.pcolormesh(*args, **kwargs)

  File "/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/matplotlib/axes.py", line 7766, in pcolormesh

    t = t._as_mpl_transform(self.axes)

  File "lib/cartopy/_crs.pyx", line 213, in cartopy._crs.CRS._as_mpl_transform (lib/cartopy/_crs.c:4071)

ValueError: Axes should be an instance of GeoAxes, got <class 'matplotlib.axes.AxesSubplot'>
```